### PR TITLE
New Package: libretls

### PR DESCRIPTION
**Pull request by Yonle**
Signed-off-by: Yonle <yonle@duck.com>

### CODE REVIEW

File: packages/libretls/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+TERMUX_PKG_HOMEPAGE=https://git.causal.agency/libretls/about/
+TERMUX_PKG_DESCRIPTION="libtls for OpenSSL"
+TERMUX_PKG_LICENSE="BSD"
```
Comment (xtkoba) [inline packages/libretls/build.sh:3]: The license for the main part of `libretls` is not `BSD` but `ISC`. I would extract the copyright information from the source code, as follows:

```
 * Copyright (c) 2014 Jeremie Courreges-Anglas <jca@openbsd.org>
 * Copyright (c) 2014 Joel Sing <jsing@openbsd.org>
 * Copyright (c) 2014 Ted Unangst <tedu@openbsd.org>
 * Copyright (c) 2015 Bob Beck <beck@openbsd.org>
 * Copyright (c) 2015 Joel Sing <jsing@openbsd.org>
 * Copyright (c) 2015 Marko Kreen <markokr@gmail.com>
 * Copyright (c) 2015 Reyk Floeter <reyk@openbsd.org>
 * Copyright (c) 2016 Bob Beck <beck@openbsd.org>
 * Copyright (c) 2016 Tobias Pape <tobias@netshed.de>
 * Copyright (c) 2021 Eric Faurot <eric@openbsd.org>
 *
 * Permission to use, copy, modify, and distribute this software for any
 * purpose with or without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 *
 * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
```

File: packages/libretls/build.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+TERMUX_PKG_HOMEPAGE=https://git.causal.agency/libretls/about/
+TERMUX_PKG_DESCRIPTION="libtls for OpenSSL"
+TERMUX_PKG_LICENSE="ISC"
+# There's no license file in the tarball source
+# Instead, copy README.7 as it also includes what LICENSE that they use.
+TERMUX_PKG_LICENSE_FILE="README.7"
```
Comment (xtkoba) [inline packages/libretls/build.sh:6]: `README.7` is not qualified as `TERMUX_PKG_LICENSE_FILE` for `ISC` license as it does not contain copyright information.

Comment (xtkoba) [reply packages/libretls/build.sh:6]: See also https://github.com/termux/termux-packages/pull/9623#discussion_r837710469.

Comment (Yonle) [reply packages/libretls/build.sh:6]: so remove that line?

Comment (xtkoba) [reply packages/libretls/build.sh:6]: Yep. And then prepare a custom `LICENSE` file in the source file tree.

File: packages/libretls/build.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+TERMUX_PKG_MAINTAINER="@Yonle <yonle@duck.com>"
+TERMUX_PKG_VERSION=3.5.1
+TERMUX_PKG_SRCURL=https://git.causal.agency/libretls/snapshot/libretls-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=dfd1fcf0015b9e99529a9d3c3cf0e4c35b7d778f120a7acc84fedd5ce13bc1f1
+TERMUX_PKG_DEPENDS="openssl"
+TERMUX_PKG_REPLACES="libtls"
+TERMUX_PKG_PROVIDEs="libtls"
```
Comment (xtkoba) [inline packages/libretls/build.sh:13]: Typo. Should be `TERMUX_PKG_PROVIDES`.

Comment (Yonle) [reply packages/libretls/build.sh]: Thanks for the information

File: packages/libretls/build.sh (hunk 1 above)
Comment (Yonle) [reply packages/libretls/build.sh]: Probably i will put a LICENSE file into the build directory.

**Comment by Yonle** [comment]
sorry for the delay. will do some stuff right now

File: packages/libretls/build.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+TERMUX_PKG_DEPENDS="openssl"
+TERMUX_PKG_REPLACES="libtls"
+TERMUX_PKG_PROVIDES="libtls"
+
+termux_step_pre_configure() {
+	autoreconf -fi
+}
+
+termux_step_install_license() {
+	install -Dm600 $TERMUX_PKG_BUILDER_DIR/LICENSE ${TERMUX_PREFIX}/share/doc/$TERMUX_PKG_NAME
```
Comment (xtkoba) [inline packages/libretls/build.sh:17]: This installs the license file as `share/doc/libretls`. This does not conform to our standard. Instead, `share/doc/libretls` should be a directory and the license file should be installed in it.

**Review by xtkoba — CHANGES_REQUESTED** [review]

File: packages/libretls/build.sh (hunk 4 above)
Comment (Yonle) [reply packages/libretls/build.sh:17]: So `install -Dm600 $TERMUX_PKG_BUILDER_DIR/LICENSE ${TERMUX_PREFIX}/share/doc/${TERMUX_PKG_NAME}/` ? Or `mkdir` first?

Comment (xtkoba) [reply packages/libretls/build.sh:17]: Use `-t` option:

```bash
install -Dm600 $TERMUX_PKG_BUILDER_DIR/LICENSE -t ${TERMUX_PREFIX}/share/doc/${TERMUX_PKG_NAME}
```

Comment (Yonle) [reply packages/libretls/build.sh:17]: I see i see

**Review by xtkoba — APPROVED** [review]

**Pull request merged**